src/compiler/program.ts

```diff
@@ -2266,8 +2266,13 @@ namespace ts {
 
             let redirectedPath: Path | undefined;
             if (refFile) {

```

Comment:
What is `refFile` in this function?

Reply:
refFile is normally passed in if its trying to resolve imports/ or reference directives etc. thats the file in which import exists and hence the file is being added. The root files passed to program, when processed have refile as undefined.